crates/multi_buffer/src/multi_buffer.rs

```diff
@@ -3052,6 +3052,15 @@ impl MultiBufferSnapshot {
         self.has_conflict
     }
 
+    pub fn has_diagnostics(&self) -> bool {
+        for excerpt in self.excerpts.iter() {
```

Comment:
NIT: could be one-liner: `self.excerpts.iter().any(|e| e.buffer.has_diagnostics())`